sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -205,6 +205,21 @@ abstract class QueryTest extends PlanTest {
         planWithCaching)
   }
 
+  /**
+   * Asserts that a given [[Dataset]] will be executed using the given named cache.
+   */
+  def assertCached(query: Dataset[_], cachedName: String): Unit = {
```

Comment:
it only checks the cache name is preserved, shall we check storage level too?

Reply:
Right. Let me make it cover the cache level as well. I am sorry that I will not have Internet access for a week. Will work on it next week.

Reply:
Hi @cloud-fan , I just updated related test case to cover storage level. Would you mind to review this PR again? Many thanks.

Reply:
Hi @cloud-fan, appreciate if you can take a look this PR again. I already updated the test case to cover storage level as well. Many thanks. Best regards, William